### ISSUE TITLE
NPM Install fail  

### ISSUE THREAD

**Issue by CitizenDev** [Shopify/js-buy-sdk#255]
Hi!

I am really interested to try out this SDK in order to implement Shopify cart right into out landing page but I got stuck while setting up test environment locally. I'm not experienced in node.js or npm.
At first I had an SSL auth issue which I resolved by replacing SSL directory with HTTP in package.json but not the npm install got stuck at "npm run build" and I haven't been able to find any information about how to resolve it.

I have included npm-debug log.

Any help would be really appreciated.
Thanks.
[npm-debug.txt](https://github.com/Shopify/js-buy-sdk/files/554799/npm-debug.txt)

**Comment by mikkoh** [comment]
Hey @CitizenDev glad you're interested in the JS Buy SDK. For your project did you install the JS Buy SDK with this command:
```bash
npm install shopify-buy --save
```

If you do this there should be no reason to run `npm run build`. Running the above command will download a prebuilt version of the JS Buy SDK from NPM. Before you run the above command make sure your project has a `package.json`. You can create a blank `package.json` file by running:
```
npm init
```

**Comment by CitizenDev** [comment]
Thanks for quick response.

But this did not resolve my issue because I still need to install the SDK-s dependencies with "npm install" command in order to run examples locally.
Let me tell you what I was planning to do exactly maybe that will help you to get me in right direction.
I was trying to get the SDK run locally and serve the examples. Because my JS skills are limited but with your documentation I would be able to edit your example code to work with with my website. If everything works locally I would include it to my website.

So If you can help me get your cart and checkout examples running I should be able to figure things out from there.

Sorry if I missed something obvious.

**Comment by NirmalBajpai** [comment]
Hey @mikkoh  I have install shopify-buy SDK using npm install shopify-buy --save
 into a new folder . Please tell me what is next steps to test cart and checkout examples with my shopify domain . I have shopify domain ApiKey and Product key of my shopify account. Please tell me I have basic knowledge about Node.js , and web development.

**Comment by NirmalBajpai** [comment]
This is my package.json file code

'
{
  "name": "shopweb",
  "version": "1.0.0",
  "description": "Backend Shop",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "Nirmal",
  "license": "ISC",
  "dependencies": {
    "shopify-buy": "^0.4.1"
  }
}
'

**Comment by minasmart** [comment]
Hey there! If you want to play around with the examples, you have to work from this project's git repository.

First clone the repo:

```bash
git clone https://github.com/Shopify/js-buy-sdk.git
cd js-buy-sdk
```

Then follow the directions [here](https://github.com/Shopify/js-buy-sdk/tree/master/examples) to get the examples up and running.

Hope this helps!!

**Comment by NirmalBajpai** [comment]
Hi please explain after clone repo what steps needed before hit  "npm install && npm run examples" command .  because this command gives this type errors:

Sameers-iMac:js-buy-sdk studiouse$ npm install && npm run examples
npm WARN package.json Dependency 'whatwg-fetch' exists in both dependencies and devDependencies, using 'whatwg-fetch@0.10.1' from dependencies
npm ERR! git clone --template=/Users/studiouse/.npm/_git-remotes/_templates --mirror ssh://git@github.com/Shopify/yuidoc-lucid-theme.git /Users/studiouse/.npm/_git-remotes/ssh-git-github-com-Shopify-yuidoc-lucid-theme-git-797ae186: Cloning into bare repository '/Users/studiouse/.npm/_git-remotes/ssh-git-github-com-Shopify-yuidoc-lucid-theme-git-797ae186'...
npm ERR! git clone --template=/Users/studiouse/.npm/_git-remotes/_templates --mirror ssh://git@github.com/Shopify/yuidoc-lucid-theme.git /Users/studiouse/.npm/_git-remotes/ssh-git-github-com-Shopify-yuidoc-lucid-theme-git-797ae186: Permission denied (publickey).
npm ERR! git clone --template=/Users/studiouse/.npm/_git-remotes/_templates --mirror ssh://git@github.com/Shopify/yuidoc-lucid-theme.git /Users/studiouse/.npm/_git-remotes/ssh-git-github-com-Shopify-yuidoc-lucid-theme-git-797ae186: fatal: Could not read from remote repository.
npm ERR! git clone --template=/Users/studiouse/.npm/_git-remotes/_templates --mirror ssh://git@github.com/Shopify/yuidoc-lucid-theme.git /Users/studiouse/.npm/_git-remotes/ssh-git-github-com-Shopify-yuidoc-lucid-theme-git-797ae186: 
npm ERR! git clone --template=/Users/studiouse/.npm/_git-remotes/_templates --mirror ssh://git@github.com/Shopify/yuidoc-lucid-theme.git /Users/studiouse/.npm/_git-remotes/ssh-git-github-com-Shopify-yuidoc-lucid-theme-git-797ae186: Please make sure you have the correct access rights
npm ERR! git clone --template=/Users/studiouse/.npm/_git-remotes/_templates --mirror ssh://git@github.com/Shopify/yuidoc-lucid-theme.git /Users/studiouse/.npm/_git-remotes/ssh-git-github-com-Shopify-yuidoc-lucid-theme-git-797ae186: and the repository exists.
npm ERR! Darwin 16.0.0
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "install"
npm ERR! node v4.5.0
npm ERR! npm  v2.15.9
npm ERR! code 128

npm ERR! Command failed: git clone --template=/Users/studiouse/.npm/_git-remotes/_templates --mirror ssh://git@github.com/Shopify/yuidoc-lucid-theme.git /Users/studiouse/.npm/_git-remotes/ssh-git-github-com-Shopify-yuidoc-lucid-theme-git-797ae186
npm ERR! Cloning into bare repository '/Users/studiouse/.npm/_git-remotes/ssh-git-github-com-Shopify-yuidoc-lucid-theme-git-797ae186'...
npm ERR! Permission denied (publickey).
npm ERR! fatal: Could not read from remote repository.
npm ERR! 
npm ERR! Please make sure you have the correct access rights
npm ERR! and the repository exists.
npm ERR! 
npm ERR! 
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/studiouse/Desktop/sdkbuy/js-buy-sdk/npm-debug.log
Sameerss-iMac:js-buy-sdk studiouse$

**Comment by minasmart** [comment]
This looks like and NPM and Node issue. Have you tried working with NPM 3+ and Node Current (Or at least 5+)?

Our distributed library works with older versions of node, but developing and running examples requires close to current versions of both Node and NPM.

Hope this helps!

**Comment by ktliu** [comment]
I have a similar issue as NirmalBajpai. Both npm and node versions are 7.2.0 and 3.10.9 respectively. 

npm ERR! Linux 4.4.0-45-generic
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "install"
npm ERR! node v7.2.0
npm ERR! npm  v3.10.9
npm ERR! code 128

**Comment by minasmart** [comment]
Hey there! Is there more to the error than that? And are you installing this module as a dependency, or are you running `npm install` inside a clone of this repository?

**Comment by ktliu** [comment]
Hi, I'm running npm install && npm run examples inside a clone of the repository

Here's the full error, thanks for the response

npm ERR! git clone --template=/home/ktliu/.npm/_git-remotes/_templates --mirror ssh://git@github.com/Shopify/yuidoc-lucid-theme.git /home/ktliu/.npm/_git-remotes/git-ssh-git-github-com-Shopify-yuidoc-lucid-theme-git-c9c62ba-fd25eb1a: Cloning into bare repository '/home/ktliu/.npm/_git-remotes/git-ssh-git-github-com-Shopify-yuidoc-lucid-theme-git-c9c62ba-fd25eb1a'...
npm ERR! git clone --template=/home/ktliu/.npm/_git-remotes/_templates --mirror ssh://git@github.com/Shopify/yuidoc-lucid-theme.git /home/ktliu/.npm/_git-remotes/git-ssh-git-github-com-Shopify-yuidoc-lucid-theme-git-c9c62ba-fd25eb1a: Permission denied (publickey).
npm ERR! git clone --template=/home/ktliu/.npm/_git-remotes/_templates --mirror ssh://git@github.com/Shopify/yuidoc-lucid-theme.git /home/ktliu/.npm/_git-remotes/git-ssh-git-github-com-Shopify-yuidoc-lucid-theme-git-c9c62ba-fd25eb1a: fatal: Could not read from remote repository.
npm ERR! git clone --template=/home/ktliu/.npm/_git-remotes/_templates --mirror ssh://git@github.com/Shopify/yuidoc-lucid-theme.git /home/ktliu/.npm/_git-remotes/git-ssh-git-github-com-Shopify-yuidoc-lucid-theme-git-c9c62ba-fd25eb1a: 
npm ERR! git clone --template=/home/ktliu/.npm/_git-remotes/_templates --mirror ssh://git@github.com/Shopify/yuidoc-lucid-theme.git /home/ktliu/.npm/_git-remotes/git-ssh-git-github-com-Shopify-yuidoc-lucid-theme-git-c9c62ba-fd25eb1a: Please make sure you have the correct access rights
npm ERR! git clone --template=/home/ktliu/.npm/_git-remotes/_templates --mirror ssh://git@github.com/Shopify/yuidoc-lucid-theme.git /home/ktliu/.npm/_git-remotes/git-ssh-git-github-com-Shopify-yuidoc-lucid-theme-git-c9c62ba-fd25eb1a: and the repository exists.
npm ERR! Linux 4.4.0-45-generic
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "install"
npm ERR! node v7.2.0
npm ERR! npm  v3.10.9
npm ERR! code 128

npm ERR! Command failed: git clone --template=/home/ktliu/.npm/_git-remotes/_templates --mirror ssh://git@github.com/Shopify/yuidoc-lucid-theme.git /home/ktliu/.npm/_git-remotes/git-ssh-git-github-com-Shopify-yuidoc-lucid-theme-git-c9c62ba-fd25eb1a
npm ERR! Cloning into bare repository '/home/ktliu/.npm/_git-remotes/git-ssh-git-github-com-Shopify-yuidoc-lucid-theme-git-c9c62ba-fd25eb1a'...
npm ERR! Permission denied (publickey).
npm ERR! fatal: Could not read from remote repository.
npm ERR! 
npm ERR! Please make sure you have the correct access rights
npm ERR! and the repository exists.
npm ERR! 
npm ERR! 
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>

npm ERR! Please include the following file with any support request:
npm ERR!     /home/ktliu/Desktop/shop/js-buy-sdk/npm-debug.log

**Comment by minasmart** [comment]
I think there was a period of time where NPM was creating invalid git+ssh references. Fixed in #278. If this doesn't fix your issue, please re-open.
